### PR TITLE
Implement option arguments

### PR DESCRIPTION
**Pull request by jeanlaroche**
Not sure whether we want the command-line arguments to override the config flags or not. But this is a start.

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
I think we can come up with a better design. I'll do a preparatory commit for it, then wave at you.

File: pythran/config.py [hunk 1]
```diff
@@ -163,7 +181,9 @@ def parse_define(define):
             extension['include_dirs'].append(numpy.get_include())
 
         # blas dependency
-        user_blas = cfg.get('compiler', 'blas')
+        user_blas = get_from_ext('compiler.blas')
```
Comment (serge-sans-paille) [inline pythran/config.py]: this shouldn't be needed if you updated the config correctly using the args

File: pythran/config.py [hunk 2]
```diff
@@ -149,7 +157,17 @@ def parse_define(define):
 
     for k, w in extra.items():
         extension[k].extend(w)
-    if cfg.getboolean('pythran', 'complex_hook'):
+
```
Comment (serge-sans-paille) [inline pythran/config.py]: Instead of adding the complexity here, I think you should udpate ``init_cfg`` to accept the extra config argument. And as you asked previously: they are more important than any config file.

Comment (jeanlaroche) [reply pythran/config.py]: OK I"ll try that.

Comment (jeanlaroche) [reply pythran/config.py]: mmm maybe I'm missing something but init_cfg(sys_file, platform_file, user_file) is called at module load time 
```
cfg = init_cfg('pythran.cfg',
               'pythran-{}.cfg'.format(sys.platform),
               '.pythranrc')
```
A that point, I don't see how I can pass the extra config arguments.

Comment (jeanlaroche) [reply pythran/config.py]: OK Take a look at it now. I think you're right, it's a lot cleaner.

File: pythran/config.py [hunk 3]
```diff
@@ -53,6 +53,21 @@ def init_cfg(sys_file, platform_file, user_file):
...
+    if config_args is not None:
+        # Override the config options with those provided on the command line e.g. compiler.blas=pythran-openblas
+        for arg in config_args:
+            try:
+                lhs,rhs = arg.split('=')
+                section,item = lhs.split('.')
+                if not cfgp.has_section(section):
+                    cfgp.add_section(section)
+                cfgp.set(section,item,rhs)
+                cfgp._sections[section][item]=rhs
```
Comment (serge-sans-paille) [inline pythran/config.py]: this looks redundant

File: pythran/config.py [hunk 4]
```diff
@@ -53,6 +53,21 @@ def init_cfg(sys_file, platform_file, user_file):
         cfgp.read([required])
     cfgp.read([user_config_path])
 
+    if config_args is not None:
+        # Override the config options with those provided on the command line e.g. compiler.blas=pythran-openblas
+        for arg in config_args:
+            try:
+                lhs,rhs = arg.split('=')
```
Comment (serge-sans-paille) [inline pythran/config.py]: please run pep8 on your modification :-)

File: pythran/config.py [hunk 5]
```diff
@@ -104,6 +119,11 @@ def lint_cfg(cfgp, **paths):
 
 def make_extension(python, **extra):
 
+    # load platform specific configuration then user configuration
+    cfg = init_cfg('pythran.cfg',
+                   'pythran-{}.cfg'.format(sys.platform),
+                   '.pythranrc',extra.get('config',None))
```
Comment (serge-sans-paille) [inline pythran/config.py]: this looks better with one line per argument (yes that's neat-picking, but the logic looks good)

**Comment by serge-sans-paille** [comment]
App veyor complains on your change, you need to remove the config argument once it's processed

File: pythran/config.py (hunk 3 above)
Comment (jeanlaroche) [reply pythran/config.py]: oops

File: pythran/config.py (hunk 4 above)
Comment (jeanlaroche) [reply pythran/config.py]: ah yes!

File: pythran/config.py [hunk 6]
```diff
@@ -13,9 +13,9 @@
 
 
 def get_paths_cfg(
-    sys_file='pythran.cfg',
-    platform_file='pythran-{}.cfg'.format(sys.platform),
-    user_file='.pythranrc'
+        sys_file='pythran.cfg',
+        platform_file='pythran-{}.cfg'.format(sys.platform),
+        user_file='.pythranrc'
```
Comment (serge-sans-paille) [inline pythran/config.py]: this diff is unrelated to this commit, right?

File: pythran/config.py [hunk 7]
```diff
@@ -53,6 +53,19 @@ def init_cfg(sys_file, platform_file, user_file):
         cfgp.read([required])
     cfgp.read([user_config_path])
 
+    if config_args is not None:
+        # Override the config options with those provided on the command line e.g. compiler.blas=pythran-openblas
+        for arg in config_args:
+            try:
+                lhs, rhs = arg.split('=')
```
Comment (serge-sans-paille) [inline pythran/config.py]: you should specify `maxsplit=1` here in case we pass `compiler.cxxflags=-fsanitize=address` or similar stuff

**Comment by serge-sans-paille** [comment]
Thanks for the contribution @jeanlaroche o/
